ldk/node/src/hostClients/browserClient.test.ts

```diff
@@ -0,0 +1,156 @@
+      await expect(queryResult).resolves.toBe(sentResponse.getUrl());
+    });
+
+    it('should call grpc client function', async () => {
+      expect(mockGRPCClient.browserActiveURL).toHaveBeenCalledWith(
+        expect.any(Messages.BrowserActiveURLRequest),
+        expect.any(Function),
+      );
+    });
+
+    it('should have attached the initial connection session to the request', () => {
+      expect(sentRequest.getSession()?.toObject()).toStrictEqual(session);
+    });
+  });
+
+  describe('#querySelectedText', () => {
+    let sentRequest: Messages.BrowserSelectedTextRequest;
+    let sentResponse: Messages.BrowserSelectedTextResponse;
+    let queryResult: Promise<BrowserSelectedTextResponse>;
+
+    beforeEach(async () => {
+      sentResponse = new Messages.BrowserSelectedTextResponse();
+
+      mockGRPCClient.browserSelectedText.mockImplementation(
+        createCallbackHandler(sentResponse),
+      );
+
+      queryResult = subject.querySelectedText();
+
+      sentRequest = captureMockArgument(mockGRPCClient.browserSelectedText);
+    });
+
+    it('should return a transformed response', async () => {
+      const selectedText = {
+        text: sentResponse.getText(),
+        url: sentResponse.getUrl(),
+        tabTitle: sentResponse.getTabtitle(),
+      };
+
+      await expect(queryResult).resolves.toStrictEqual(selectedText);
+    });
+
+    it('should call grpc client function', async () => {
+      expect(mockGRPCClient.browserSelectedText).toHaveBeenCalledWith(
+        expect.any(Messages.BrowserSelectedTextRequest),
+        expect.any(Function),
+      );
+    });
+
+    it('should have attached the initial connection session to the request', () => {
+      expect(sentRequest.getSession()?.toObject()).toStrictEqual(session);
+    });
+  });
+
+  describe('#streamActiveURL', () => {
+    let sentRequest: Messages.BrowserActiveURLStreamRequest;
+
+    beforeEach(async () => {
+      mockGRPCClient.browserActiveURLStream.mockImplementation(
+        createStreamingHandler(),
+      );
+
+      subject.streamActiveURL(identityCallback);
+
+      sentRequest = captureMockArgument(mockGRPCClient.browserActiveURLStream);
+    });
+
+    it('should have attached the initial connection session to the request', () => {
+      expect(sentRequest.getSession()?.toObject()).toStrictEqual(session);
+    });
+  });
+
+  describe('#streamSelectedText', () => {
+    let sentRequest: Messages.BrowserSelectedTextStreamRequest;
+
+    beforeEach(async () => {
+      mockGRPCClient.browserSelectedTextStream.mockImplementation(
+        createStreamingHandler(),
+      );
+
+      subject.streamSelectedText(identityCallback);
+
+      sentRequest = captureMockArgument(
+        mockGRPCClient.browserSelectedTextStream,
+      );
+    });
+
+    it('should have attached the initial connection session to the request', () => {
+      expect(sentRequest.getSession()?.toObject()).toStrictEqual(session);
```

Comment:
I wondered if the safe navigation operator was introducing a false negative risk here, but since we're assigning `session` in the before each block, this shouldn't be an issue (and probably stops TS from complaining, though the not-null assertion operator would've worked too I think)

Reply:
Yeah, I can triple check that if `?.` does return early that it still throws.

Reply:
Actually, for these purposes, I'll just do a not-null check. If there had been a `matchesInterface` that might have been a good middle ground.

Reply:
Addressed